torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -518,7 +519,69 @@ class CudaKernelGenerator : private kir::IrVisitor {
     return cast.str();
   }
 
+  // If possible, replace pow with mul. Return true when successful.
+  bool genPowerWithMul(const kir::BinaryOp* node) {
+    if (node->operation() != BinaryOpType::Pow) {
+      return false;
+    }
+
+    auto rhs = node->rhs();
+    c10::optional<double> exponent;
+    if (auto val_int = dynamic_cast<kir::Int*>(rhs)) {
+      if (val_int->isConst()) {
+        exponent = val_int->value().value();
+      }
+    } else if (auto val_float = dynamic_cast<kir::Double*>(rhs)) {
+      if (val_float->isConst()) {
+        auto fp_exp = val_float->value().value();
+        double int_exp = 0;
+        if (std::modf(fp_exp, &int_exp) == 0) {
+          exponent = int_exp;
+        }
+      }
+    }
+
+    if (!exponent.has_value()) {
+      return false;
+    }
+
+    // Only **2 and **3 are considered
+    if (!(exponent.value() == 2 || exponent.value() == 3)) {
+      return false;
+    }
+
+    auto lhs = gen(node->lhs());
+
+    if (print_inline_) {
+      code_ << lhs << " * " << lhs;
+      if (exponent.value() == 3) {
+        code_ << " * " << lhs;
+      }
+    } else {
+      indent() << gen(node->out());
+      if (node->out()->isScalar()) {
+        code_ << " = " << lhs << " * " << lhs;
```

Comment:
Is there a `exponent.value() == 3` case here?

Reply:
Oops, thanks. Fixed. I also extended the test to exercise this code path.